### DISCUSSION TITLE
Add extentions for the pigpio daemon to set options

### DISCUSSION THREAD

**Discussion by paulvee** [Poeschl/Hassio-Addons#299] [Ideas]
Hi Poeschl!
Thank you for the many extensions and add-ons.

With the upcoming deprecation of RPi.GPIO I'm suspecting that there will be more demand for the pigpio module.
I am using Pyscipt to create a Python script that uses pigpio to drive a PWM fan.

Because the pigpio daemon is currently invoked without the possibility of adding commands, we're stuck with the default 5uS sample speed, which limits the PWM frequency to 8KHz. Some fans, like the Noctua ones, require 25KHz. With a 1uS sample rate, we can get close enough - 20KHz, if we could invoke option "-s 1". While you're at it, we should be able to add any of the options.

Thanks in advance.
Paul

**Comment by Poeschl** [comment]
That will be a nice addition 💡 indeed. If you want feel free to add it and contribute.
For myself I'm currently short on free time for developing.

**Comment by paulvee** [comment]
Hi Markus,
I would love to contribute, unfortunately I'm very green in the HA environment and do not know how to do this.

I already looked very hard for details and went through the Github details but can't even find the location where the pigpio stuff goes on my RPi installation.
If you can get me some pointers I will attempt it.

Paul

**Reply by paulvee** [reply]
I looked at your github files once again and it seems you could brute force an additional option to the daemon by forking the project and then changing the Run file:
Hassio-Addons/tree/master/pigpio/root/etc/services.d/pigpio/run

The last line now has : "pigpiod -g -f"   and can be changed to "pigpiod -g -f -s 1" to get a 1uS sample rate.
In order to invoke this, you would have to remove the pigpio installation from HA and then install it again with the modified run file to get the new invocation loaded.

**Comment by Poeschl** [comment]
I did some changes to the addon, so when #303 is merged, additional arguments can be provided via the addon options.

**Reply by paulvee** [reply]
Super Markus! Thank you
